### PR TITLE
Makefile: Add coverage-report target

### PR DESCRIPTION
**Pull request by mgold1234**
This commit introduces a new 'coverage-report' target to the Makefile, enhancing the developer's ability to analyze code coverage directly from the terminal.

The new target allows developers to:
- Generate a comprehensive coverage report for the entire project

This addition streamlines the code coverage review process, enabling developers to quickly focus on particular areas of interest within the codebase.

### CODE REVIEW

**Review by schuellerf — APPROVED** [review]
nice, thanks

**Review by schuellerf — CHANGES_REQUESTED** [review]
Oh - overlooked this, please add `coverage.html` to `.gitignore`

File: Makefile [hunk 1]
```diff
@@ -14,6 +14,8 @@ help:
 	@echo "    dev-prerequisites:  Install necessary development prerequisites on your system"
 	@echo "    push-check:         Replicates the github workflow checks as close as possible"
 	@echo "                        (do this before pushing!)"
+	@echo "	   coverage-report:    Run unit tests and generate an HTML coverage report."
+	@echo "    coverage-dump:     Run unit tests and display function-level coverage information."
```
Comment (schuellerf) [inline Makefile:18]: tiny nit-pick? but I'll approve anyway
```suggestion
	@echo "    coverage-dump:      Run unit tests and display function-level coverage information."
```

**Review by schuellerf — DISMISSED** [review]

**Review by schuellerf — CHANGES_REQUESTED** [review]
`coverage.out` should not be part of the git commit, I guess
although this is not mentioned in the `Makefile` maybe also add it to `.gitignore` (after removing from git)

File: Makefile [hunk 2]
```diff
@@ -60,6 +62,18 @@ dev-prerequisites:
...
+.PHONY: coverage-dump
+coverage-dump: unit-tests
+	go tool cover -func=coverage.txt
+
+.PHONY: coverage-report
+coverage-report: unit-tests
+	go tool cover -o coverage.html -html coverage.txt
+
+.PHONY: coverage-dump
+coverage-dump: unit-tests
```
Comment (schuellerf) [inline Makefile:74]: This target seems to be duplicated?

**Review by schuellerf — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (mvo5) [inline Makefile:74]: It looks like this stanza is a duplicate of the one in line 65. We probably only need one "coverage-dump" target or am I missing something? (copy/paste issue maybe?)

File: .gitignore [hunk 3]
```diff
@@ -10,4 +10,6 @@ dnf-json
 /gen-oscap
 local.env
 __debug*
-coverage.txt
+coverage*
+
```
Comment (mvo5) [inline .gitignore:14]: (super nitick) extra newlines here that we probably don't need (but they also don't hurt so fine to ignore)

**Review by mvo5 — COMMENTED** [review]
Thank you for this PR. I added a small comment/question inline, once the double "coverage-dump" is clarified/fixed this is good to go from my PoV

File: Makefile (hunk 2 above)
Comment (mgold1234) [reply Makefile]: thanks

**Review by mvo5 — APPROVED** [review]
Thank you!

**Review by schuellerf — APPROVED** [review]
Thank you!

**Pull request merged**